### PR TITLE
FIX-#3605: Address comments related to new project structure

### LINKED ISSUE #3605: Address comments related to new project structure
We should address comments in merge #3537.

### PR DESCRIPTION
**Pull request by YarShev**
Signed-off-by: Igoshev, Yaroslav <yaroslav.igoshev@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #3605  
- [x] tests passing
- [x] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/core/dataframe/base/partitioning/axis_partition.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+        Returns
+        -------
+        BaseDataframeAxisPartition
+            An axis partition containing only a single materialized partition.
+        """
+        materialized = self.apply(
+            lambda x: x, num_splits=1, maintain_partitioning=False
+        )
+        return type(self)(materialized, get_ip=get_ip)
```
Comment (vnlitvinov) [inline modin/core/dataframe/base/partitioning/axis_partition.py:128]: why do we have an implementation in abstract base?..

File: modin/core/dataframe/base/partitioning/axis_partition.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        list
+            List of partitions from this axis partition.
+
+        Notes
+        -----
+        If `get_ip=True`, a list of tuples of Ray.ObjectRef/Dask.Future to node ip addresses and
+        unwrapped partitions, respectively, is returned if Ray/Dask is used as an engine
+        (i.e. [(Ray.ObjectRef/Dask.Future, Ray.ObjectRef/Dask.Future), ...]).
+        """
+        if squeeze and len(self.list_of_blocks) == 1:
```
Comment (vnlitvinov) [inline modin/core/dataframe/base/partitioning/axis_partition.py:152]: same, it's an abstract base, why implement anything in it?

Comment (YarShev) [reply modin/core/dataframe/base/partitioning/axis_partition.py:152]: It's legacy but yes we want to keep abstract base classes with methods declaration only. Should probably be fixed as part of a separate issue.

File: modin/core/dataframe/base/partitioning/axis_partition.py (hunk 1 above)
Comment (YarShev) [reply modin/core/dataframe/base/partitioning/axis_partition.py:128]: It's legacy but yes we want to keep abstract base classes with methods declaration only. Should probably be fixed as part of a separate issue.

Comment (vnlitvinov) [reply modin/core/dataframe/base/partitioning/axis_partition.py:128]: Could you please make a separate issue then?

**Review by vnlitvinov — APPROVED** [review]
LGTM

File: modin/core/dataframe/base/partitioning/axis_partition.py (hunk 1 above)
Comment (YarShev) [reply modin/core/dataframe/base/partitioning/axis_partition.py:128]: Sure, #3635

**Pull request merged**